drivers/hmis/lib/hmis_data_cleanup/util.rb

```diff
@@ -0,0 +1,137 @@
+###
+# Copyright 2016 - 2023 Green River Data Analysis, LLC
+#
+# License detail: https://github.com/greenriver/hmis-warehouse/blob/production/LICENSE.md
+###
+
+# Utility functions for cleaning up HMIS data during a migration.
+# Be careful, some functions modify data and don't leave a paper trail.
+#
+# Usage: HmisDataCleanup::Util.assign_missing_household_ids!
+module HmisDataCleanup
+  class Util
+    include ArelHelper
+
+    # Assign Household ID where missing
+    def self.assign_missing_household_ids!
+      scope = Hmis::Hud::Enrollment.hmis.where(household_id: nil)
+      Rails.logger.info "Assigning household id to #{scope.size} enrollments"
+      scope.each do |enrollment|
+        hh_id = Hmis::Hud::Base.generate_uuid
+        without_papertrail_or_timestamps do
+          enrollment.update_columns(household_id: hh_id) # skips callbacks
```

Comment:
If this gets to be a large number, and upsert is probably worth doing.  You could just `scope.find_in_batches` and then `enrollment.assign_attributes` and finally "import" the batch with a conflict key of `id`.